File: fixtures/scale/dart/medium/lib/io/result_printer.dart
import '../config/settings.dart';
import 'output_formatter.dart';

class ResultPrinter {
  final OutputFormatter formatter;

  ResultPrinter(Settings settings) : formatter = OutputFormatter(settings);

  void printValue(double value) {
    print('= ${formatter.format(value)}');
  }
}

File: fixtures/scale/java/medium/src/calc/io/ResultPrinter.java
package calc.io;

import java.io.PrintStream;

import calc.config.Settings;

public class ResultPrinter {
    private final OutputFormatter formatter;

    public ResultPrinter(Settings settings) {
        this.formatter = new OutputFormatter(settings);
    }

    public void print(PrintStream out, double value) {
        out.println("= " + formatter.format(value));
    }
}
